### DISCUSSION TITLE
Text expansion is not working when terminal based editors are opened in kitty

### DISCUSSION THREAD

**Discussion by fromZahir2Aleph** [kovidgoyal/kitty#6585] [Q&A]
I use text expansion app, espanso. It works fine with the exception of terminal based editors like nano and helix in a kitty terminal. In case of a single line returns wrongly formatted text. E.g. instead of "Hi there!" "hi there1". And instead of multi-line just a single letter "v". The issue is present only within kitty whatever shell I use and whichever editor I open.

**Comment by kovidgoyal** [comment]
You will need to contact the developers of expanso. Presumably whatever hack they are using to inject text is broken.

**Reply by fromZahir2Aleph** [reply]
It doesn't work with abbreviations function of  Autokey either. Output has wrong formatting in terminal based editors as well as command line itself. Again I tried different shells and different terminal emulators. The problem occurs only in kitty whatever shell I use and whichever editor I open.

**Reply by kovidgoyal** [reply]
Yes all these programs use all manner of unholy hacks to do what they do.  I'm afraid I dont have any interest in debugging them. https://github.com/espanso/espanso/issues/281
